### ISSUE TITLE
Prototype a substrate-neutral bounded structured-read envelope

### ISSUE THREAD

**Issue by rinsakamo** [rinsakamo/relay-self#135]
## Summary

Define and prototype the smallest **substrate-neutral bounded structured-read envelope** needed by RelayEngine experiments, informed by the public behavior of `mmastrac/djev-spark` without copying its source.

Parent: #101.

This Issue is an **experiment-interface / falsification ledger**. It is not semantic authority, execution authority, or permission to add a permanent provider abstraction before evidence requires one.

The target is not to adopt djev-spark itself. The target is to preserve one useful property:

> RelaySelf should be able to ask a bounded typed question of different cognition substrates without changing the semantic ownership of Present, Skill, THINK, Action, or World truth.

## Provenance / license boundary

Reference implementation:

- https://github.com/mmastrac/djev-spark
- public README documents a Jev-compatible `POST /v1/systemone` surface over DiffusionGemma structured reads.

At the time this Issue was opened, GitHub reports **no repository license** for `mmastrac/djev-spark`.

Therefore:

- do not copy source code, tests, templates, or substantial implementation text from djev-spark;
- use only publicly observable API/behavioral ideas as design inspiration;
- implement any RelaySelf experiment code independently from current RelaySelf authority and requirements;
- if upstream later adds a license, re-evaluate separately rather than retroactively assuming permission.

## Useful ideas to test

The public djev-spark surface demonstrates several ideas relevant to #101/#84:

1. **typed bounded reads**
   - yes/no;
   - finite choice;
   - ordered score;

2. **probability/confidence-shaped diagnostics**
   - choice probabilities where the substrate exposes meaningful logits;
   - confidence / entropy as diagnostics, not truth;

3. **adaptive extra reads**
   - one cheap read first;
   - spend additional samples/steps only when uncertainty remains high;

4. **dependent staged questions**
   - later bounded questions need not run unless earlier answers make them relevant;

5. **explicit usage/diagnostics**
   - expose cognition cost and read structure separately from the semantic answer.

These are implementation ideas, not RelaySelf semantic owners.

## Critical RelaySelf difference: THINK stays outside

djev-spark exposes an optional `think` budget inside the decision request.

Do **not** copy that boundary into RelaySelf 1.0.

Current #87 treats THINK as a special Skill / slow-cognition escape path. Preserve:

~~~text
bounded structured read
  -> adequate
     -> continue bounded path

  -> uncertain / novel / contradictory / unresolved
     -> RelaySelf may select or invoke THINK
~~~

not:

~~~text
provider decides to hide free-form Thinking inside a bounded read
~~~

This distinction keeps cognition allocation visible and lets THINK remain replaceable, costed, and governed by Self rather than by one provider.

## Candidate experiment-local envelope

Start with an experiment-local typed shape. Do not promote it into a repository-wide runtime API yet.

Conceptually:

~~~text
request:
  projected_state
  bounded_questions[]
  cognition_budget

bounded_question:
  id
  kind = boolean | choice | ordered_score
  admissible_answers / criteria

result:
  answers[]
  resolution = resolved | unresolved
  diagnostics
  usage
~~~

Where available, diagnostics may include:

~~~text
candidate probabilities
entropy / margin
samples / denoising steps / NFE
native mode
parse/canonicalization source
latency / token / forward counts
~~~

But preserve:

~~~text
probability != truth
confidence != authority
resolved != Action authorization
model answer != World fact
~~~

## Mapping to current owners

The envelope must not create a second state authority.

~~~text
Present Projection / Skill-local projection
  -> supplies the smallest sufficient cognition-facing state

RelayEngine experiment
  -> asks bounded typed question(s)

substrate
  -> AR / dLM / Linear Self-Spec / future DiffusionGemma structured read / deterministic control

result
  -> transient decision evidence

existing RelaySelf owners
  -> decide selection, THINK escalation, SkillExecution, Action authorization, consequence handling
~~~

Refs #105 for the input projection seam.

## Conditional questions

The djev-spark `depends_on` / `ask_if` shape is useful, but do not create a generic dependency language in RelaySelf yet.

First test the reduction:

~~~text
question 1
  -> result materially narrows question 2 candidate set
  -> question 2 is admitted only if still relevant
~~~

using ordinary decision-epoch orchestration.

Example:

~~~text
hazard present?
  -> no  -> do not ask obstacle-side question
  -> yes -> ask obstacle-side question
~~~

This is evidence for cognitive narrowing only if it preserves outcome quality while reducing cost.

## Adaptive-read rule

A provider may support:

~~~text
one read
  -> adequate / stable
     -> stop

  -> uncertain
     -> additional sample / step
~~~

But the stopping rule must be measured per substrate.

Do not assume:

- first-step diffusion logits are calibrated probabilities;
- AR token probabilities and diffusion read probabilities are directly comparable;
- entropy thresholds transfer across models;
- repeated samples are always cheaper than THINK escalation.

Any uncertainty threshold remains experiment-local until calibrated evidence supports reuse.

## First matched use

After #133 produces terminal NLD cost evidence, use one already-qualified bounded subject and place at least two mechanisms behind the same experiment-local envelope:

~~~text
A. current NLD bounded decision path
B. simplest alternative bounded mechanism available locally
   (deterministic/fake provider for interface tests first;
    later DiffusionGemma structured-read provider if physically qualified)
~~~

The first goal is not cross-model superiority.

It is to test:

> Can the same RelaySelf bounded decision meaning survive a substrate swap without changing Present ownership, Skill semantics, THINK escalation, or Action/World authority?

## Required deterministic tests before physical comparison

Test at least:

1. boolean / choice / ordered-score validation;
2. invalid candidate rejection;
3. unresolved result is representable without forcing a winner;
4. provider diagnostics cannot mutate semantic state;
5. conditional second question is skipped when irrelevant;
6. conditional second question executes when relevant;
7. cost/usage is reported separately from answer semantics;
8. THINK escalation is outside the provider envelope;
9. stale Present / #104 reopening can invalidate a prior bounded result;
10. two fake providers with different native diagnostics produce the same canonical bounded answer shape.

## Grand Null

Do not add merely for this experiment:

- a permanent `SystemOneClient`;
- a new top-level `Decision` owner;
- a provider registry;
- a generic dependency DSL;
- hidden provider-owned THINK;
- a universal confidence threshold;
- an assumption that all substrates expose probabilities;
- an OpenAI/Jev compatibility endpoint in the product runtime;
- a hard dependency on djev-spark, vLLM, or DiffusionGemma.

If current local experiment types are sufficient, keep the seam local.

## Relationship to current work

- #101 owns adaptive cognition substrate experiments.
- #133 currently owns the fresh matched-adequacy NLD cost surface and should terminate before this Issue claims a physical provider comparison.
- #84 owns the broader crystallized/fast-cognition hypothesis.
- #87 owns Skill-centered narrowing and THINK as a slow-cognition escape path.
- #105 owns Present Projection -> RelayEngine input semantics.
- #104 owns reopening after contradictory World consequence.
- #9 remains literature provenance; djev-spark is an engineering reference rather than a paper.

## Completion

This Issue completes when:

1. a minimal experiment-local canonical bounded-read shape exists;
2. deterministic tests demonstrate that at least two provider implementations can satisfy it without semantic-owner changes;
3. THINK escalation remains outside the provider;
4. one already-qualified bounded decision subject is replayed through the envelope after #133 terminal evidence;
5. the result is classified as either:
   - useful substrate-exchange seam;
   - unnecessary abstraction reducible to existing experiment code;
   - insufficient because providers require materially incompatible semantics.

Only a repeated non-reducible need should justify promotion into durable RelaySelf runtime authority.

Refs #101.
Refs #133.
Refs #84.
Refs #87.
Refs #105.
Refs #104.

**Comment by rinsakamo** [comment]
MVP reconciliation: #135 is useful but **not a prerequisite** for the first RelaySelf 1.0 vertical slice in #136.

#140 may use an experiment-local bounded request/result shape if that is the smallest correct integration. Do not force a permanent substrate-neutral provider abstraction into the MVP merely to satisfy #135.

If #140 independently demonstrates repeated non-reducible provider interchange requirements, that evidence can return here.

Refs #136.
Refs #140.
